src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6144,27 +6134,21 @@ public void SetProperty(string path, PSObject propertyToSet)
 
             PSObject results = new PSObject();
             PSObject fileSystemInfoShell = null;
-            bool isContainer = false;
 
-            // Create a PSObject with either a DirectoryInfo or FileInfo object
-            // at its core.
+            var fsinfo = GetFileSystemInfo(path, out bool isDirectory);
 
-            if (Utils.ItemExists(path, out bool isDirectory))
+            // Create a PSObject with either a DirectoryInfo or FileInfo object at its core.
+            if (fsinfo != null)
             {
                 if (isDirectory)
                 {
-                    isContainer = true;
-                    fileSystemInfoShell = PSObject.AsPSObject(new DirectoryInfo(path));
+                    fileSystemInfoShell = PSObject.AsPSObject((DirectoryInfo)fsinfo);
                 }
                 else
                 {
-                    // Maybe the path is a file name so try a FileInfo instead
-                    fileSystemInfoShell = PSObject.AsPSObject(new FileInfo(path));
+                    fileSystemInfoShell = PSObject.AsPSObject((FileInfo)fsinfo);
```

Comment:
ditto.
Actually, this if/else block can be removed.

Reply:
Fixed.